File: Leetcode/0102. Binary Tree Level Order Traversal /102.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> arr =new ArrayList<>();
        List<Integer> level = new ArrayList<>();
        Queue <TreeNode> q = new LinkedList<>();
        if(root == null)
            return arr;
        q.add(root);
        q.add(null);
        while(!q.isEmpty())
        {
            
            TreeNode temp = q.remove();
            System.out.print(temp+" "+ q.size());
            if(temp == null)
            {
                arr.add(level);
                if(q.isEmpty())
                    return arr;
                level = new ArrayList<>();
                q.add(null);
            }
            else
            {
                level.add(temp.val);
                if(temp.left != null)
                    q.add(temp.left);
                if(temp.right != null)
                    q.add(temp.right);
            }
        }
        return arr;
    }
}

File: Leetcode/0102. Binary Tree Level Order Traversal /102.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        arr = []
        q = deque()
        if(root == None):
            return arr
        q.append(root)
        q.append(None)
        arr1 = []
        while(len(q) != 0):
            temp = q.popleft()
            if(temp == None):
            
                arr.append(arr1)
                arr1 = []
                if(len(q) ==0):
                    return arr
                q.append(None)
            else:
                arr1.append(temp.val)
                if(temp.left != None):
                    q.append(temp.left)
                if(temp.right != None):
                    q.append(temp.right)
            print(len(q))
        return arr


